### PR TITLE
LPS-186190 - Add support to Info Framework for executing actions

### PR DESCRIPTION
**Pull request by ruben-pulido**
//cc @jkappler @ealonso @lfbesada @sandrodw3

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by ruben-pulido** [comment]
ci:test:sf

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by ruben-pulido** [comment]
ci:test:sf

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by ruben-pulido** [comment]
I've updated [baseline commit](https://github.com/liferay-echo/liferay-portal/pull/12256/commits/956c8d6006214e01040a2747feba21dccc2ccb68) to fix [Semantic Versioning Test failure](https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/504576/testReport/com.liferay.semantic.versioning/SemanticVersioningTest/testSemanticVersioning__apps_info_info_api_/)

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by Tim-Cao** [comment]
Hi @ruben-pulido, faliures are unrelated to your changes

File: modules/apps/info/info-api/src/main/java/com/liferay/info/item/InfoItemActionExecutor.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.liferay.info.item;
+
+import com.liferay.info.exception.InfoItemActionExecutionException;
+
+/**
+ * @author Rubén Pulido
+ */
+public interface InfoItemActionExecutor<T> {
```
Comment (ealonso) [inline modules/apps/info/info-api/src/main/java/com/liferay/info/item/InfoItemActionExecutor.java:22]: Hey @ruben-pulido, What do you think about moving this interface to a new package called executor or action.executor?

File: modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/ExecuteInfoItemActionStrutsAction.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception);
+			}
+
+			InfoItemActionExecutionException infoItemActionExecutionException =
+				new InfoItemActionExecutionException();
+
+			if (exception instanceof PrincipalException) {
```
Comment (ealonso) [inline modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/ExecuteInfoItemActionStrutsAction.java:111]: @ruben-pulido it is always false

File: modules/apps/info/info-api/src/main/java/com/liferay/info/item/InfoItemActionExecutor.java (hunk 1 above)
Comment (ruben-pulido) [reply modules/apps/info/info-api/src/main/java/com/liferay/info/item/InfoItemActionExecutor.java:22]: Makes total sense. In my head I had done that, but obviously I didn't :|

File: modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/ExecuteInfoItemActionStrutsAction.java (hunk 2 above)
Comment (ruben-pulido) [reply modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/request/struts/ExecuteInfoItemActionStrutsAction.java:111]: I'll remove this if

**Comment by ruben-pulido** [comment]
Feedback incorporated here: https://github.com/liferay-echo/liferay-portal/pull/12289

**Pull request closed without merging**